format/postgres/pg_control.go

```diff
@@ -0,0 +1,170 @@
+package postgres
+
+import (
+	"embed"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/postgres/common"
+	"github.com/wader/fq/format/postgres/flavours/pgpro10"
+	"github.com/wader/fq/format/postgres/flavours/pgpro11"
+	"github.com/wader/fq/format/postgres/flavours/pgpro12"
+	"github.com/wader/fq/format/postgres/flavours/pgpro13"
+	"github.com/wader/fq/format/postgres/flavours/pgpro14"
+	"github.com/wader/fq/format/postgres/flavours/pgproee10"
+	"github.com/wader/fq/format/postgres/flavours/pgproee11"
+	"github.com/wader/fq/format/postgres/flavours/pgproee12"
+	"github.com/wader/fq/format/postgres/flavours/pgproee13"
+	"github.com/wader/fq/format/postgres/flavours/pgproee14"
+	"github.com/wader/fq/format/postgres/flavours/pgproee15"
+	"github.com/wader/fq/format/postgres/flavours/postgres10"
+	"github.com/wader/fq/format/postgres/flavours/postgres11"
+	"github.com/wader/fq/format/postgres/flavours/postgres12"
+	"github.com/wader/fq/format/postgres/flavours/postgres13"
+	"github.com/wader/fq/format/postgres/flavours/postgres14"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+)
+
+//go:embed pg_control.md
+var pgControlFS embed.FS
+
+func init() {
+	interp.RegisterFormat(format.PG_CONTROL, &decode.Format{
+		Description: "PostgreSQL control file",
+		DecodeFn:    decodePgControl,
+		DefaultInArg: format.PgControlIn{
+			Flavour: "",
+		},
+	})
+	interp.RegisterFS(pgControlFS)
+}
+
+const (
+	PG_CONTROL_VERSION_10    = 1002
+	PG_CONTROL_VERSION_11    = 1100
+	PGPRO_CONTROL_VERSION_11 = 1200
+	PG_CONTROL_VERSION_12    = 1201
+	//PG_CONTROL_VERSION_13 = 1300
+	PG_CONTROL_VERSION_14 = 1300
+)
+
+const (
+	PG_FLAVOUR_POSTGRES10 = "postgres10"
+	PG_FLAVOUR_POSTGRES11 = "postgres11"
+	PG_FLAVOUR_POSTGRES12 = "postgres12"
+	PG_FLAVOUR_POSTGRES13 = "postgres13"
+	PG_FLAVOUR_POSTGRES14 = "postgres14"
+	PG_FLAVOUR_POSTGRES15 = "postgres15"
+	PG_FLAVOUR_PGPRO10    = "pgpro10"
+	PG_FLAVOUR_PGPRO11    = "pgpro11"
+	PG_FLAVOUR_PGPRO12    = "pgpro12"
+	PG_FLAVOUR_PGPRO13    = "pgpro13"
+	PG_FLAVOUR_PGPRO14    = "pgpro14"
+	PG_FLAVOUR_PGPRO15    = "pgpro15"
+	PG_FLAVOUR_PGPROEE10  = "pgproee10"
+	PG_FLAVOUR_PGPROEE11  = "pgproee11"
+	PG_FLAVOUR_PGPROEE12  = "pgproee12"
+	PG_FLAVOUR_PGPROEE13  = "pgproee13"
+	PG_FLAVOUR_PGPROEE14  = "pgproee14"
+	PG_FLAVOUR_PGPROEE15  = "pgproee15"
+)
+
+func decodePgControl(d *decode.D) any {
+	d.Endian = decode.LittleEndian
+
+	var pgIn format.PgControlIn
+	if !d.ArgAs(&pgIn) {
+		d.Fatalf("DecodeInArg must be PgControlIn!\n")
```

Comment:
Maybe update error message, something like "no flavour specified"? btw is it possible to probe flavour if not set?

There are some other d.ArgAs cases where message need update

Reply:
It's done.